### DISCUSSION TITLE
better way to set img src file for an avatar?

### DISCUSSION THREAD

**Discussion by solarslurpi** [executablebooks/meta#878] [Q&A] [answered]
Hello,
- I extended my sphinx environment to include [sphinx-design].
- 
I am using [avatars](https://sphinx-design.readthedocs.io/en/furo-theme/css_classes.html#avatars).

I write sphinx docs in markdown.

I usually set an image like:
```
```{image} images/ebp-logo.png
:align: left
:scale: 20
```
using i`mages/ebg-logo.png` works for the path.

the only way to get the avatar to find the image was to use a full path.  e.g.:
```{grid-item-card}
<img src="http://127.0.0.1:8000/_images/ebp-logo.png" class="sd-avatar-md sd-border-2">
```
Is there something I am missing/doing wrong?  Thank you.

**Comment by NickleDave** [comment]
Can you say a little more about what you mean by "full path"?  
Does it work if you remove "http://127.0.0.1:8000/"?  

The examples in the docs use paths that are written relative to the docs directory, e.g. 
https://github.com/executablebooks/sphinx-design/blob/fd7912f5da5f681daee17bbc7a6bcee117f006a8/docs/css_classes.md?plain=1#L282

``````markdown
````{grid} 1 2 3 3
:gutter: 1
```{grid-item-card} sd-avatar-xs
<img src="images/ebp-logo.png" class="sd-avatar-xs sd-border-1">
```
```{grid-item-card} sd-avatar-sm
<img src="images/ebp-logo.png" class="sd-avatar-sm sd-border-1">
```
```{grid-item-card} sd-avatar-md
<img src="images/ebp-logo.png" class="sd-avatar-md sd-border-1">
```
```{grid-item-card} sd-avatar-lg
<img src="images/ebp-logo.png" class="sd-avatar-lg sd-border-1">
```
```{grid-item-card} sd-avatar-xl
<img src="images/ebp-logo.png" class="sd-avatar-xl sd-border-1">
```
````

``````

where those images live in
https://github.com/executablebooks/sphinx-design/tree/fd7912f5da5f681daee17bbc7a6bcee117f006a8/docs/images

but this is something I have found myself confused by before -- it can depend on how the rest of your project is set up (I think)

**Reply by solarslurpi** [reply]
Thank you for the reply.

I started with 
:::{div}
<img src="images/ebp-logo.png class="sd-avatar-md sd-border-1">
:::

This doesn't work.  I looked at google dev's inspector and the path in the html was just images/edp-logo.png (hence 404).

When I did the image tag, e.g.:
```{image} images/ebp-logo.png
:align: center
:scale: 20

```
this works fine (all I need is a spiffy circle around it :-) ).

This got me thinking (I have been told I am not that smart - so warning there)... that css happens after sphinx resolves all path goodness like with {image} directive.  I just want a {div} class="sd-avatar-..' to reolve image paths like the {image} directive.

From what I can tell, I have the same sphinx setup that I see here.  All images resolve correctly.  Just not avatars.  Again, thank you.

**Comment by solarslurpi** [comment]
and hence my confusion.  Why do the examples work?

**Comment by NickleDave** [comment, marked as answer]
:thinking: I could be wrong but I think the secret sauce is enabling the myst extension `html_image` like so:
https://github.com/executablebooks/sphinx-design/blob/fd7912f5da5f681daee17bbc7a6bcee117f006a8/docs/conf.py#L56

If this is the project you're working on, then I think you don't have it enabled, right?
https://github.com/solarslurpi/GrowBuddy/blob/1734c1b870f6ebd121df046461ae6991d8adb60e/docs/conf.py#L41

See https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#images for more info on what's going on here

edit: adding context from that part of the myst parser docs:

> The final option is directly using HTML, which is also parsed by MyST. This is usually a bad option, because the HTML is treated as raw text during the build process and so sphinx will not recognise that the image file is to be copied, and will not output the HTML into non-HTML output formats.

So your intuition is right I think that you need to do a little extra work here so Sphinx can resolve things correctly

**Reply by solarslurpi** [reply]
That worked!  Thank you very much.  Right there in the documentation.  My apologies.  Perhaps it would be helpful to provide a link to this documentation within the css page of [sphinx-design] ?  Thank you for your kind replies.  Many smiles!

**Reply by NickleDave** [reply]
🫡 happy to help, agree an admonition pointing out html_image in the avatar section of sphinx-design docs might be good -- @solarslurpi maybe you could raise an issue there suggesting it and link this discussion? Since you spotted the need for it :slightly_smiling_face:

**Reply by solarslurpi** [reply]
I will give it a go.  Again, thank you.  I found the spot in the documentation where I believe it should be added.
